### ISSUE TITLE
When SDL used as subproject, SDL_revision.h regenerates after every commit

### ISSUE THREAD

**Issue by 1vanK** [libsdl-org/SDL#7561]
I use SDL as subproject of game engine. After every commit to the engine SDL's cmake config gets the hash of this commit and update SDL_revision.h. This cause unnecessary SDL compilation.

```
if(EXISTS "${CMAKE_CURRENT_SOURCE_DIR}/VERSION.txt")
  # If VERSION exists, it contains the SDL version
  file(READ "${CMAKE_CURRENT_SOURCE_DIR}/VERSION.txt" SDL_REVISION_CENTER)
  string(STRIP "${SDL_REVISION_CENTER}" SDL_REVISION_CENTER)
else()
  # If VERSION does not exist, use git to calculate a version
  git_describe(SDL_REVISION_CENTER)
  if(NOT SDL_REVISION_CENTER)
    set(SDL_REVISION_CENTER "${SDL3_VERSION}-no-vcs")
  endif()
endif()
```

Can you add option to set SDL_REVISION_CENTER without creating VERSION.txt file and using the git command?

Or just

```
if(NOT PREDEFINED_REVISION_CENTER)
  if(EXISTS "${CMAKE_CURRENT_SOURCE_DIR}/VERSION.txt")
    ...
  else()
    # If VERSION does not exist, use git to calculate a version
    ..
  endif()
endif()
```

**Comment by madebr** [comment]
It's working as intended to make sure `SDL_GetRevision` returns the correct value.
When it detects a changed git checkout, it rebuilds the bare minimum (`src/SDL.c`) and relinks SDL.
The CMake revision logic expects SDL to be either a release, or checked out from git as a submodule.

How are you using SDL as a subproject? `FetchContent`? Copied SDL's sources?

The cost of relinking your (c++) application on every commit can be high, so I'll look at making it configurable.

**Comment by 1vanK** [comment]
SDL sources copied to subfolder and added to source three by `add_subdirectory()`

https://github.com/dviglo/dviglo/tree/main/source/third-party/sdl

**Comment by madebr** [comment]
This has been fixed in #8030.
Thanks!
